File: backend/src/main/java/bg/pcbudget/backend/models/Keyboard.java
package bg.pcbudget.backend.models;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "keyboards")
@Getter
@Setter
@NoArgsConstructor
public class Keyboard extends Product {

  @ManyToOne
  @JoinColumn(name = "interface_id")
  private ItemInterface interfaceType;

  private String connectionType;
  private Boolean hasBulgarianLayout;
  private String keyboardType;
  private String color;
  private String switches;
  private Boolean hasLighting;
}

File: src/app/models/Keyboard.ts
import { Product } from './Product';
import { Interface } from './Interface';

export interface Keyboard extends Product {
  interfaceType:      Interface;
  connectionType:     string;
  hasBulgarianLayout: boolean;
  keyboardType:       string;
  color:              string;
  switches:           string;
  hasLighting:        boolean;
}
